### ISSUE TITLE
LUCENE SPATIAL corrupts OrientDb when query out of range  (V.2.1 release)

### ISSUE THREAD

**Issue by MVdev1** [orientechnologies/orientdb#4880]
Executing a query with a bad value can corrupt LUCENE, making the database unusable for this feature until rebuild/restart.

Example of a bad query...

SELECT y, x, in(_l0).size() AS n FROM L0 WHERE [y, x] WITHIN [[-45, 135], [-8
    9.99, **1444**]]

REPRO:

I've got some tables that use integer fields or double fields (same result).

```
    /*** Location stuff - e.g. Lat/Long ***/
    CREATE CLASS LL EXTENDS V ABSTRACT;
        CREATE PROPERTY LL.y DOUBLE;
        ALTER PROPERTY LL.y NOTNULL true;
        ALTER PROPERTY LL.y MANDATORY true;
        CREATE PROPERTY LL.x DOUBLE;
        ALTER PROPERTY LL.x NOTNULL true;
        ALTER PROPERTY LL.x MANDATORY true;

    CREATE CLASS L1 EXTENDS LL;
        CREATE INDEX L1.l ON L1(y,x) SPATIAL ENGINE LUCENE;

    CREATE CLASS L2 EXTENDS LL;
        CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE;

    CREATE CLASS L0 EXTENDS V;
        CREATE PROPERTY L0.y Short;
        ALTER PROPERTY L0.y NOTNULL true;
        ALTER PROPERTY L0.y MANDATORY true;
        CREATE PROPERTY L0.x Short;
        ALTER PROPERTY L0.x NOTNULL true;
        ALTER PROPERTY L0.x MANDATORY true;

        CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE;
```

I can rebuild everything and use it (usually) then calling with an out of range function from OrientJs shows an error that my query was out of range.  I can rebuild the indexes as suggested but still get the same result (it doesn't work) **even with good queries.**

```
2015-08-28 20:33:37:418 INFO  {db=TestDb} - Rebuilding index TestDb.L0.l (
estimated 893 items)... [OIndexRebuildOutputListener]
2015-08-28 20:33:38:008 INFO  {db=TestDb} --> **OK, indexed 893 items in 590 ms
 [OIndexRebuildOutputListener]**
2015-08-28 20:33:39:824 INFO  {db=TestDb} - Rebuilding index TestDb.L1.l (
estimated 3,094 items)... [OIndexRebuildOutputListener]
2015-08-28 20:33:40:534 INFO  {db=TestDb} --> OK, indexed 3,094 items in 710
ms [OIndexRebuildOutputListener]
2015-08-28 20:33:46:077 INFO  {db=TestDb} - Rebuilding index TestDb.L2.l (
estimated 4,583 items)... [OIndexRebuildOutputListener]
2015-08-28 20:33:46:571 INFO  {db=TestDb} --> OK, indexed 4,583 items in 494
ms [OIndexRebuildOutputListener]{db=TestDb} Error on using index L0.l in quer
y 'SELECT y, x, in(_l0).size() AS n FROM L0 WHERE [y, x] WITHIN [[-45, 135], [-8
9.99, 179.99]] ORDER BY n DESC LIMIT 15'. Probably you need to rebuild indexes.
Now executing query using cluster scan
com.spatial4j.core.exception.InvalidShapeException: maxY must be >= minY: -45.0
to -89.99
        at com.spatial4j.core.context.SpatialContext.makeRectangle(SpatialContex
t.java:220)
        at com.spatial4j.core.context.SpatialContext.makeRectangle(SpatialContex
t.java:205)
        at com.orientechnologies.lucene.shape.ORectangleShapeFactory.makeShape(O
RectangleShapeFactory.java:42)
        at com.orientechnologies.lucene.shape.OShapeFactoryImpl.makeShape(OShape
FactoryImpl.java:44)
        at com.orientechnologies.lucene.manager.OLuceneSpatialIndexManager.searc
hWithin(OLuceneSpatialIndexManager.java:176)
        at com.orientechnologies.lucene.manager.OLuceneSpatialIndexManager.get(O
LuceneSpatialIndexManager.java:126)
        at com.orientechnologies.lucene.OLuceneIndexEngine.get(OLuceneIndexEngin
e.java:115)
        at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.get(OLuceneI
ndexNotUnique.java:110)
        at com.orientechnologies.lucene.index.OLuceneIndexNotUnique.get(OLuceneI
ndexNotUnique.java:36)
        at com.orientechnologies.orient.core.index.OIndexAbstractDelegate.get(OI
ndexAbstractDelegate.java:59)
        at com.orientechnologies.orient.core.index.OIndexTxAwareMultiValue.get(O
IndexTxAwareMultiValue.java:272)
        at com.orientechnologies.orient.core.index.OIndexTxAwareMultiValue.get(O
IndexTxAwareMultiValue.java:40)
        at com.orientechnologies.lucene.operator.OLuceneWithinOperator.executeIn
dexQuery(OLuceneWithinOperator.java:69)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searc
hForIndexes(OCommandExecutorSQLSelect.java:1859)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searc
hInClasses(OCommandExecutorSQLSelect.java:850)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbs
tract.assignTarget(OCommandExecutorSQLResultsetAbstract.java:207)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.assig
nTarget(OCommandExecutorSQLSelect.java:475)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execu
teSearch(OCommandExecutorSQLSelect.java:457)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execu
te(OCommandExecutorSQLSelect.java:428)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.exe
cute(OCommandExecutorSQLDelegate.java:90)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginat
edStorage.executeCommand(OAbstractPaginatedStorage.java:1468)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginat
edStorage.command(OAbstractPaginatedStorage.java:1449)
        at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract
.execute(OCommandRequestTextAbstract.java:63)
        at com.orientechnologies.orient.server.network.protocol.http.command.pos
t.OServerCommandPostCommand.execute(OServerCommandPostCommand.java:81)
        at com.orientechnologies.orient.server.network.protocol.http.ONetworkPro
tocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:180)
        at com.orientechnologies.orient.server.network.protocol.http.ONetworkPro
tocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:627)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:
77)
```

Error on using index L0.l in query '**SELECT y, x, in(_l0).size() AS n FROM L0 WHERE [y, x] WITHIN [[-45, 135], [-89.99, 179.99]] ORDER BY n DESC LIMIT 15**'. Probably you need to rebuild indexes.    Now executing query using cluster scan

Another strange thing is that I'm usually able to continue to run this query in studio but not in Javascript and it usually starts working again once I've completely shutdown the server and brought it back up.

```
    var sql = "SELECT y, x, in(_l0).size() AS n FROM L0 WHERE [y, x] WITHIN [[-45, 135], [-8
9.99, 179.99]] ORDER BY n DESC LIMIT 15";
db.query(sql)
    .then(function (results)
    {
        console.log(JSON.stringify(results));
    });
```

And it makes no different if I put LIMIT in sql or as a parameter.

I had seen something in the log which said that one of my index files was locked but that might have been from other Lucene tables that I deleted.

Finally, I loaded with declare intent bulk upload so maybe that caused this issue?

**Comment by seeden** [comment]
I had a same experience with corrupted lucent index 
+1 for fix

**Comment by MVdev1** [comment]
More info....  (maybe helpful)

Regarding the LUCENE SPATIAL INDEX on a table of lat/long data where the values are shorts representing just the degrees and I see...

2015-08-29 00:49:38:465 INFO  OrientDB Server v2.1.0 (build UNKNOWN@r${buildNumber};

2015-08-05 13:38:23+0000) is active. [OServer]{db=TestDb} **Error on using index L0.y_x in query 'SELECT y, x FROM L0 WHERE [y, x] WITHIN [[-119, 33], [-117,31]] ORDER BY n DESC LIMIT 50'. Probably you need to rebuild indexes. Now executing query using cluster scan**

**java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.Short**
        at com.orientechnologies.common.serialization.types.OShortSerializer.preprocess(OShortSerializer.java:36)
        at com.orientechnologies.orient.core.serialization.serializer.binary.impl.index.OCompositeKeySerializer.preprocess(OCompositeKeySerializer.java:383)
        at com.orientechnologies.orient.core.serialization.serializer.binary.impl.index.OCompositeKeySerializer.preprocess(OCompositeKeySerializer.java:46)
        at com.orientechnologies.orient.core.index.hashindex.local.OLocalHashTable.get(OLocalHashTable.java:320)

When running...

SELECT y,x FROM L0 WHERE [y,x] WITHIN [[33,-119],[31,-117]] ORDER BY n DESC LIMIT 50
or
SELECT y,x FROM L0 WHERE [y,x] WITHIN [[-119,33],[-117,31]] ORDER BY n DESC LIMIT 50

*based on the documentation I'm still not clear which order to put lat/long on each side of the WITHIN -- see issue 4881*

From a table that looks like.....

    CREATE CLASS L0 EXTENDS V;
	CREATE PROPERTY L0.y Short;
	ALTER PROPERTY L0.y NOTNULL true;
	ALTER PROPERTY L0.y MANDATORY true;
	CREATE PROPERTY L0.x Short;
	ALTER PROPERTY L0.x NOTNULL true;
	ALTER PROPERTY L0.x MANDATORY true;
	
	CREATE INDEX L0.i ON L0(y,x) UNIQUE_HASH_INDEX;
	CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE;

PS....   [This might also be helpful.]

In this last case the index did not appear in studio (but the files are there in the 'databases' folder) so I ran the following command..

    CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE;

It came back with an error "2015-08-29 05:52:03:930 SEVER Internal server error: com.orientechnologies.orient.core.index.OIndexException: Index with type SPATIAL and algorithm LUCENE; does not exist. [ONetworkProtocolHttpDb]"

So I dropped the index...

     DROP INDEX L0.i

And created it again...

    CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE

This time I see, "2015-08-29 05:52:37:936 INFO  {db=TestDb} - Rebuilding index TestDb.L0.l (estimate
d 3,835 items)... [OIndexRebuildOutputListener]
2015-08-29 05:52:41:587 INFO  {db=TestDb} --> OK, indexed 3,835 items in 3,651 ms [OIndexRebuildOutputListener]"

This query still returns no results...

    SELECT y,x FROM L0 WHERE [y,x] WITHIN [[33,-119],[31,-117]] ORDER BY n DESC LIMIT 50

And I now see this (see after the rebuilding line)...

2015-08-29 05:52:37:936 INFO  {db=Test} - Rebuilding index Test.L0.l (estimated 3,835 items)... [OIndexRebuildOutputListener]
2015-08-29 05:52:41:587 INFO  {db=Test} --> OK, indexed 3,835 items in 3,651 ms [OIndexRebuildOutputListener]{db=Test} Error on using index L0.l in query 'SELECT y, x FROM L0 WHERE [y, x] WITHIN [[33, -119], [31, -117]] ORDER BY n DESC LIMIT 50'. Probably you need to rebuild indexes. Now executing query using cluster
 scan
com.spatial4j.core.exception.**InvalidShapeException: maxY must be >= minY: 33.0 to 31.0**
        at com.spatial4j.core.context.SpatialContext.makeRectangle(SpatialContext.java:220)
        at com.spatial4j.core.context.SpatialContext.makeRectangle(SpatialContext.java:205)
        at com.orientechnologies.lucene.shape.ORectangleShapeFactory.makeShape(ORectangleShapeFactory.java:42)
        at com.orientechnologies.lucene.shape.OShapeFactoryImpl.makeShape(OShapeFactoryImpl.java:44)
        at com.orientechnologies.lucene.manager.OLuceneSpatialIndexManager.searchWithin(OLuceneSpatialIndexManager.java:176)
        at com.orientechnologies.lucene.manager.OLuceneSpatialIndexManager.get(OLuceneSpatialIndexManager.java:126)

I drop and rebuild the LUCENE SPATIAL INDEX and run the query with the parameters in opposite order

    SELECT y,x FROM L0 WHERE [y,x] WITHIN [[-119,33],[-117,31]] ORDER BY n DESC LIMIT 50

and get this...

{db=TestDb} Error on using index L0.l in query 'SELECT y, x FROM L0 WHERE [y, x]
WITHIN [[-119, 33], [-117, 31]] ORDER BY n DESC LIMIT 50'. Probably you need to
rebuild indexes. Now executing query using cluster scan
com.spatial4j.core.exception.InvalidShapeException: **Bad Y value -119.0** is not in
 boundary Rect(minX=-180.0,maxX=180.0,minY=-90.0,maxY=90.0)
        at com.spatial4j.core.context.SpatialContext.verifyY(SpatialContext.java
:193)
        at com.spatial4j.core.context.SpatialContext.makePoint(SpatialContext.ja
va:199)
        at com.orientechnologies.lucene.shape.ORectangleShapeFactory.makeShape(O
RectangleShapeFactory.java:39)
        at com.orientechnologies.lucene.shape.OShapeFactoryImpl.makeShape(OShape
FactoryImpl.java:44)

** THE IMPORTANT PART AND THE REASON THIS IS AN IMPORTANT BUG **

Good queries don't work after a bad query...

So, if I drop the index, recreate it and run some good query then everything is fine and I get results...  e.g.

     DROP INDEX L0.l
     CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE
     SELECT y,x FROM L0 WHERE [y,x] WITHIN [[41.8,12.4],[42,14]] ORDER BY n DESC LIMIT 50

**Comment by maggiolo00** [comment]
@MVdev1  
the first error is due  the double index on the same fields one HASH_INDEX and one SPATIAL  and the operator uses the wrong one

**Comment by MVdev1** [comment]
Regarding the double index... I need it for inserts & updates.  Yes, I get that it is skipping the Lucene index but I don't understand why it works for 2 of the 3 classes and they all have basically the same data and are created in basically the same way, in the case of L1/L2 they are both derived from the same class.

**OK** - SELECT y,x,in(_l0).size() AS n FROM L0 WHERE [y,x] WITHIN [[33,-118],[34,-116]] ORDER BY n DESC LIMIT 50;
**ERROR** - SELECT y,x,in(_l1).size() AS n FROM L1 WHERE [y,x] WITHIN [[33,-118],[34,-116]] ORDER BY n DESC LIMIT 50;
**OK** - SELECT y,x,in(_l2).size() AS n FROM L2 WHERE [y,x] WITHIN [[33,-118],[34,-116]] ORDER BY n DESC LIMIT 50;

They are created in the same way...

	DROP INDEX L0.l;  DROP INDEX L0.h;
	DROP INDEX L1.l;  DROP INDEX L1.h;
	DROP INDEX L2.l;  DROP INDEX L2.h;

	CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE;
	CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE;
	CREATE INDEX L1.l ON L1(y,x) SPATIAL ENGINE LUCENE;

	CREATE INDEX L0.h ON L0(y,x) UNIQUE_HASH_INDEX;
	CREATE INDEX L1.h ON L1(y,x) UNIQUE_HASH_INDEX;
	CREATE INDEX L2.h ON L2(y,x) UNIQUE_HASH_INDEX;

**Could it be related to inheritance?**

By changing the order in which I create the indexes, I change which queries work and which fail.

L0 always works (no inheritance) but now L1 works and L2 fails.  **VERY INTERESTING, RIGHT?**

	orientdb> connect plocal:/c:/OrientDb/databases/AMIDb admin admin

	orientdb {db=AMIDb}> DROP INDEX L0.l

	orientdb {db=AMIDb}> DROP INDEX L0.h

	orientdb {db=AMIDb}> DROP INDEX L1.l

	orientdb {db=AMIDb}> DROP INDEX L1.h

	orientdb {db=AMIDb}> DROP INDEX L2.l

	orientdb {db=AMIDb}> DROP INDEX L2.h

	orientdb {db=AMIDb}> CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE

	orientdb {db=AMIDb}> CREATE INDEX L1.l ON L1(y,x) SPATIAL ENGINE LUCENE

	orientdb {db=AMIDb}> CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE

	orientdb {db=AMIDb}> CREATE INDEX L0.h ON L0(y,x) UNIQUE_HASH_INDEX

	orientdb {db=AMIDb}> CREATE INDEX L1.h ON L1(y,x) UNIQUE_HASH_INDEX

	orientdb {db=AMIDb}> CREATE INDEX L2.h ON L2(y,x) UNIQUE_HASH_INDEX

	orientdb {db=AMIDb}> SELECT y,x,in(_l0).size() AS n FROM L0 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2

	orientdb {db=AMIDb}> SELECT y,x,in(_l1).size() AS n FROM L1 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2

	orientdb {db=AMIDb}> SELECT y,x,in(_l2).size() AS n FROM L2 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2
	{db=AMIDb} Error on using index L2.h in query 'select y,x,in(_l2).size() AS n FR
	OM L2 WHERE [y,x] WITHIN [[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2'. Probabl
	y you need to rebuild indexes. Now executing query using cluster scan
	java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.Do
	uble
			at com.orientechnologies.common.serialization.types.ODoubleSerializer.pr
	eprocess(ODoubleSerializer.java:36)

This isn't my first bug found with inheritance.

The base class 'LL' is described in the top of this issue report.

And why is it that, for example, the OrientDb/databases/MyTestDb/luceneIndexes/L1.l folder is not removed when the index is deleted?

**Comment by MVdev1** [comment]
FYI - still happening in 2.1.1

	orientdb> connect plocal:/c:/OrientDb/databases/TestDb admin admin

	orientdb {db=TestDb}> DROP INDEX L0.l
	orientdb {db=TestDb}> DROP INDEX L0.h
	orientdb {db=TestDb}> DROP INDEX L1.l
	orientdb {db=TestDb}> DROP INDEX L1.h
	orientdb {db=TestDb}> DROP INDEX L2.l
	orientdb {db=TestDb}> DROP INDEX L2.h
	orientdb {db=TestDb}> CREATE INDEX L0.l ON L0(y,x) SPATIAL ENGINE LUCENE
	orientdb {db=TestDb}> CREATE INDEX L1.l ON L1(y,x) SPATIAL ENGINE LUCENE
	orientdb {db=TestDb}> CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE
	orientdb {db=TestDb}> CREATE INDEX L0.h ON L0(y,x) UNIQUE_HASH_INDEX
	orientdb {db=TestDb}> CREATE INDEX L1.h ON L1(y,x) UNIQUE_HASH_INDEX
	orientdb {db=TestDb}> CREATE INDEX L2.h ON L2(y,x) UNIQUE_HASH_INDEX

	orientdb {db=TestDb}> SELECT y,x,in(_l0).size() AS n FROM L0 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2

	orientdb {db=TestDb}> SELECT y,x,in(_l1).size() AS n FROM L1 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2

	orientdb {db=TestDb}> SELECT y,x,in(_l2).size() AS n FROM L2 WHERE [y,x] WITHIN [
	[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2
	{db=TestDb} Error on using index L2.h in query 'select y,x,in(_l2).size() AS n FR
	OM L2 WHERE [y,x] WITHIN [[33,-118],[34,-116]] ORDER BY n DESC LIMIT 2'. Probabl
	y you need to rebuild indexes. Now executing query using cluster scan
	java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.Do
	uble

**Comment by MVdev1** [comment]
So, it took nearly 16 hours to remove all the links in the data and create L1 and L2 without using inheritance and I get the same result.  The query works on L1 but not on L2.  VERY STRANGE as they are identical but one has 2 decimal places of accuracy and the other only in a float type.

One new thing with 2.1.1 is that the luceneIndexes folder now disappears when I drop all the Lucene indexes.

**And Enrico...   YOU ARE CORRECT.  Everything works if I don't also have teh UNIQUE_HASH_INDEX on the same table.**

Maybe THIS ^^^ is the real bug here?

Of course (1) I can't make updates in this table efficiently without some kind of HASH index, (2) it's strange that it works on 2 of the 3 tables with the same dual indexing and (3) it's strange that removing the HASHing didn't fix my problem when I was using inheritance from an abstract base class.

Let me know if you need help or access to my database in order to check any of these -- it's not yet in production as I'm just experimenting but I really want to put it out in a beta soon.  ;)

**Comment by maggiolo00** [comment]
Hi

MVdev1
yes the dual index on the same fields is something that we are working on.
The problem is that lucene is an external plugin
and the Sql Engine doesn't know him. But we are working on it.

Yes you can send me something so i can reproduce your issue and fix them.

Thanks
Enrico

2015-09-01 11:03 GMT+02:00 MVdev1 <notifications@github.com>:

**Comment by MVdev1** [comment]
I now have corrupted a database with 400 million edges because of this bug.  I was putting the UNIQUE_HASH back during load and taking it away during run but somehow the query didn't work and I ended up with no way to put the UNIQUE index back.

Is there a way to find all the duplicates and remove them?

Any updates on when this will be fixed?

**Comment by maggiolo00** [comment]
hi @MVdev1 

about the two indexes Hash/Lucene on the same fields?

**Comment by MVdev1** [comment]
Yes, I think this is exactly what the bug is.  I am **unable to use the Lucene index when the Hash index exists** and it is strange (as I recall) because it works for some number of queries (with both idexes applied) then stops working.

I'm not sure if it matters which order I create the indexes...

CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE
CREATE INDEX L2.h ON L2(y,x) UNIQUE_HASH_INDEX

or 

CREATE INDEX L2.h ON L2(y,x) UNIQUE_HASH_INDEX
CREATE INDEX L2.l ON L2(y,x) SPATIAL ENGINE LUCENE

The indexes create fine but then I start getting errors in the log when I use them.

- - - - 

I was trying to think of a work-around like keeping a second copy of the same table, one with Lucene indexes and another with a UNIQUE Hash index.

OR

Maybe I should be running my queries explicitly against the named indexes?

*Thoughts?*

**Comment by MVdev1** [comment]
PS  As noted from our discussion regarding 4879....

>>        Did you create another index on the same fields?
>>        Yes, the problem goes away when I have only the Lucene index, regardless of inheritance.

So my workaround is to swap indexes depending on what take I'm doing. When in insert mode then I add the HASH indexes, when in view mode I take out the HASH and add back the LUCENE.

**Comment by robfrank** [comment]
The problem of having 2 indexes on a single property is addressed on 2.2.23:
https://github.com/orientechnologies/orientdb/issues/7513
